lib/puppetserver/ca/ca.rb

```diff
@@ -0,0 +1,210 @@
+require 'puppetserver/ca/utils/http_client'
+
+require 'json'
+
+module Puppetserver
+  module Ca
+    class Ca
+
+      REVOKE_BODY = JSON.dump({ desired_state: 'revoked' })
+      SIGN_BODY =   JSON.dump({ desired_state: 'signed' })
+
+      def initialize(logger, settings)
+        @logger = logger
+        @client = Puppetserver::Ca::Utils::HttpClient.new(settings)
+        @ca_server = settings[:ca_server]
+        @ca_port = settings[:ca_port]
+      end
+
+      def sign_certs(certnames)
+        put('certificate_status', certnames, SIGN_BODY, :sign)
+      end
+
+      def revoke_certs(certnames)
+        put('certificate_status', certnames, REVOKE_BODY, :revoke)
+      end
+
+      # Should this csr be in JSON? instead of string?
```

Comment:
I'm sending the csr as a string, which works.  Should it be  JSON.dump like the other bodies?